pkg/types/helm/providence.go

```diff
@@ -0,0 +1,132 @@
+//
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helm
+
+import (
+	"bytes"
+	"io"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+
+	"golang.org/x/crypto/openpgp"
+	"golang.org/x/crypto/openpgp/armor"
+	"golang.org/x/crypto/openpgp/clearsign"
+)
+
+type Provedance struct {
+	ChartMetadata    map[string]string
+	SumCollection    *SumCollection
+	ArmoredSignature *armor.Block
+	block            *clearsign.Block
+}
+
+type SumCollection struct {
+	Files  map[string]string `json:"files"`
+	Images map[string]string `json:"images,omitempty"`
+}
+
+func (p *Provedance) Unmarshal(reader io.Reader) error {
+
+	buf := &bytes.Buffer{}
+	_, err := buf.ReadFrom(reader)
```

Comment:
nit: you could obtain the count of bytes read here rather than doing `len(rawProvenanceFile)` below

Reply:
Resolved